### ISSUE TITLE
[css-inline-3] Naming text-box-trim et al.

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#10675]
Just landed the edits to split `text-box-edge` into two properties and add a shorthand with text-box-trim in https://github.com/w3c/csswg-drafts/commit/7869dc6823e2ce9c6d816c33fb070ed5c317eb1a ; see
* [CSSWG resolution in 8829](https://github.com/w3c/csswg-drafts/issues/8829#issuecomment-1699500216)
* [Editor's Draft: line box sizing property definition](https://drafts.csswg.org/css-inline-3/#text-edges)
* [Editor's Draft: leading-trimming properties and shorthand](https://drafts.csswg.org/css-inline-3/#leading-trim)

We had left open some amount of syntax bikeshedding to follow up on, so before I publish the Editor's Draft I wanted to run the current state by the CSSWG.

**Here's the syntax overview:**
```
<text-edges> = [ text | cap | ex | ideographic | ideographic-ink ]
               [ text | alphabetic | ideographic | ideographic-ink ]?
<text-trim> = start | end | both

/* line box sizing property (inherited) */
line-fit-edge: leading | <text-edges>

/* leading-trim properties (non-inherited) */
text-box: normal | <text-trim> || <text-edges>
text-box-trim: none | <text-trim>
text-box-edge: auto | <text-edges>
```

**Shorthand details:**
* `text-box: normal` sets to initial values (`none`/`auto`).
* Setting only `<text-trim>` defaults `text-box-edge` to `auto` (initial value).
* Setting only `<text-edges>` defaults `text-box-trim` to `both` (not initial value).

**Additional details:**
* `text-box-edge: auto` copies from `line-fit-edge`, treating `leading` as `text`.
* Reminder: `line-fit-edge` values other than `leading` opt into the newer line box sizing model (leading only on root inline, honoring margin/border/padding on inline boxes).

**Consequences:**
* Defaulting this way enables trimming the leading by just setting a `<text-trim>` value.
* If the author set something more aggressive and script-specific for the inherited `line-fit-edge` property, that gets passed through as the default trim edges.
* They can alternatively set an explicit trim value in the `text-box` shorthand or cascade one independently with the `text-box-edge` longhand.
* Shorthanding magic for `text-box` means that:
    * `text-box: both` sets trimming to the default edge on both sides.
    * `text-box: cap alphabetic` sets trimming to those edges on both sides.
    * `text-box: normal` does the normal thing of not trimming the line box. :)

**We should rename the trim keywords:**
Having the `text-box` shorthand take `start | end | both` reads badly, suggest renaming these keywords to `trim-start | trim-end | trim-both` for two reasons:
* Consistency with `text-spacing-trim`
* The shorthand reads a lot more reasonably: `text-box: trim-both` rather than `text-box: both` which is weird.

**Proposed Resolutions:**
* Adopt `text-box`/`text-box-trim`/`text-box-edge` properties as outlined above.
* Rename `start | end | both` values of `text-box-trim` to `trim-start | trim-end | trim-both`.
* Name the line-box sizing property (previously discussed as `text-line-edge` to contrast with `text-box-edge`) to `line-fit-edge` (for now), or to some alternative that people might suggest here.
* Republish css-inline-3 as WD with [these edits](https://drafts.csswg.org/css-inline-3/#changes).

**Comment by nileshprajapati** [comment]
> * Rename `start | end | both` values of `text-box-trim` to `trim-start | trim-end | trim-both`.
 To be consistent with margin-trim, I would suggest renaming `trim-both` to just `trim`.

**Comment by fantasai** [comment]
> To be consistent with margin-trim, I would suggest renaming trim-both to just trim.

`margin-trim` doesn’t have a `trim` value though?

**Comment by nileshprajapati** [comment]
> > To be consistent with margin-trim, I would suggest renaming trim-both to just trim.
> 
> `margin-trim` doesn’t have a `trim` value though?

Yes, margin-trim does not have a `trim` value but does have both a `block` and `inline` values which could work in the same way.

`trim` value can mean both `trim-start` & `trim-end`.

**Margin trim example:**
...
margin-trim: block;
margin-trim: block-start;
margin-trim: block-end;

margin-trim: inline;
margin-trim: inline-start;
margin-trim: inline-end;

https://developer.mozilla.org/en-US/docs/Web/CSS/margin-trim

**Comment by kojiishi** [comment]
The proposal looks fine to me, thank you.

I'm also fine with either `trim-both` or `trim`, mildly prefer `trim`.

**Comment by fantasai** [comment]
@kojiishi Should we change `text-spacing` to match?

**Comment by yisibl** [comment]
Adding the `trim-` prefix makes it inconsistent with [white-space-trim](https://drafts.csswg.org/css-text-4/#propdef-white-space-trim) in addition to `margin-trim`.

```
white-space-trim：none | discard-before || discard-after || discard-inner
```

**Comment by fantasai** [comment]
@yisibl The behavior of `white-space-trim` is a bit different than `text-box-trim` or `text-spacing-trim`: the `discard-before` and `discard-after` values discard space _outside_ the box.
